modules/layers/src/geojson-layer/geojson-layer.js

```diff
@@ -149,43 +204,76 @@ export default class GeoJsonLayer extends CompositeLayer {
     }
   }
 
-  /* eslint-disable complexity */
+  /* eslint-disable complexity, max-statements */
   renderLayers() {
     // Layer composition props
     const {stroked, filled, extruded, wireframe, material, transitions} = this.props;
 
     // Rendering props underlying layer
     const {
-      lineWidthUnits,
-      lineWidthScale,
-      lineWidthMinPixels,
-      lineWidthMaxPixels,
-      lineJointRounded,
+      alphaCutoff,
+      background,
+      backgroundPadding,
+      billboard,
+      characterSet,
+      elevationScale,
+      fontFamily,
+      fontSettings,
+      fontWeight,
+      iconAtlas,
+      iconMapping,
       lineCapRounded,
+      lineDashJustified,
+      lineHeight,
+      lineJointRounded,
       lineMiterLimit,
-      pointRadiusUnits,
-      pointRadiusScale,
-      pointRadiusMinPixels,
-      pointRadiusMaxPixels,
-      elevationScale,
-      lineDashJustified
+      lineWidthMaxPixels,
+      lineWidthMinPixels,
+      lineWidthScale,
+      lineWidthUnits,
+      maxWidth,
+      onIconError,
+      outlineColor,
+      outlineWidth,
+      pointType,
+      sizeMaxPixels,
+      sizeMinPixels,
+      sizeScale,
+      sizeUnits,
+      wordBreak
     } = this.props;
 
     // Accessor props for underlying layers
     const {
-      getLineColor,
+      getAlignmentBaseline,
+      getAngle,
+      getBackgroundColor,
+      getBorderColor,
+      getBorderWidth,
+      getColor,
+      getElevation,
       getFillColor,
-      getRadius,
-      getLineWidth,
+      getIcon,
+      getLineColor,
       getLineDashArray,
-      getElevation,
+      getLineWidth,
+      getPixelOffset,
+      getRadius,
+      getSize,
+      getText,
+      getTextAnchor,
       updateTriggers
     } = this.props;
 
     const PolygonFillLayer = this.getSubLayerClass('polygons-fill', SolidPolygonLayer);
     const PolygonStrokeLayer = this.getSubLayerClass('polygons-stroke', PathLayer);
     const LineStringsLayer = this.getSubLayerClass('line-strings', PathLayer);
-    const PointsLayer = this.getSubLayerClass('points', ScatterplotLayer);
+
+    const PointsLayerClass = PointsLayerClassMap[pointType];
+    if (!PointsLayerClass) {
+      throw new Error(`pointType must be one of: ${Object.keys(PointsLayerClassMap)}`);
```

Comment:
Layers usually just fall back to the default if an invalid enum is supplied.

Reply:
Changed to fallback as suggested